invenio_rdm_records/stats/permissions.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 TU Wien.
+#
+# Invenio RDM Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Permission factories for Invenio-Stats.
+
+In contrast to the very liberal defaults provided by Invenio-Stats, these permission
+factories deny access unless otherwise specified.
+"""
+
+from flask import g
+from invenio_stats import current_stats
+
+from ..proxies import current_rdm_records_service as records_service
+
+
+class StatsPermissionTranslator:
+    """Translator class between the statistics endpoint and the service layer."""
+
+    def __init__(self, action_name, service=None, **params):
+        """Constructor."""
+        self.service = service or records_service
+        self.action_name = action_name
+        self.params = params
+
+    def translate_params(self, identity):
```

Comment:
The `identity` argument is accepted but not used here because this `StatsPermissionTranslator` class is expected to be subclassed to match various needs.
If we're not passing the identity here, subclasses would also have to access `flask.g.identity`, which feels a bit weird and requires an additional import.

For example, I can easily imagine a use-case where the record needs to be resolved from the DB (e.g. via the service) before further checks.

This could be done by code like:
```python
def translate_params(self, identity):
  params = {**self.params}
  if "recid" in params:
    rec = self.service.read(id_=params["recid"], identity=identity)._record
    params["record"] = rec  # for the permission generators

  return params
```